### ISSUE TITLE
Block validation console errors for jetpack/subscriptions

### ISSUE THREAD

**Issue by adimoldovan** [Automattic/jetpack#26753]
### Impacted plugin

Jetpack

### Steps to Reproduce

On Atomic:
1. Enable Subscriptions: wp-admin/admin.php?page=jetpack#/settings?term=subscriptions, enable all 3 options.
2. Add or edit a post and check the browser's console output

### A clear and concise description of what you expected to happen.

No errors.

### What actually happened

Errors are displayed.

```
Block validation: Block validation failed for `jetpack/subscriptions` ({name: 'jetpack/subscriptions', icon: {…}, keywords: Array(3), attributes: {…}, providesContext: {…}, …}).

Content generated by `save` function:

<div class="wp-block-jetpack-subscriptions wp-block-jetpack-subscriptions__supports-newline wp-block-jetpack-subscriptions__use-newline">[jetpack_subscription_form subscribe_placeholder="Email Address" show_subscribers_total="false" button_on_newline="true" custom_background_button_color="#000000" custom_text_button_color="#ffffff" custom_font_size="16px" custom_border_radius="0" custom_border_weight="1" custom_padding="15" custom_spacing="10" submit_button_classes="" email_field_classes="" show_only_email_and_button="true" success_message="Success! An email was just sent to confirm your subscription. Please find the email now and click &#039;Confirm Follow&#039; to start subscribing."]</div>

Content retrieved from post body:

<div class="wp-block-jetpack-subscriptions wp-block-jetpack-subscriptions__supports-newline wp-block-jetpack-subscriptions__use-newline">[jetpack_subscription_form subscribe_placeholder="Email Address" show_subscribers_total="false" button_on_newline="true" custom_background_button_color="#000000" custom_text_button_color="#ffffff" custom_font_size="16px" custom_border_radius="0" custom_border_weight="1" custom_padding="15" custom_spacing="10" submit_button_classes="" email_field_classes="" show_only_email_and_button="true"]</div>
```

### Other information

On simple sites, or If "subscribe to site” and option "subscribe to comments” are disabled the same error is displayed but for `core/button`  

```
Block validation: Block validation failed for `core/button` ...
```

### Platform (Simple, Atomic, or both?)

Simple, Atomic

### Reproducibility

Consistent

**Comment by csabarakasz** [comment]
These two might be related: https://github.com/Automattic/wp-calypso/issues/69604

**Comment by jeherve** [comment]
This should be fixed as of #26947
